pkg/couchdb/couchdb.go

```diff
@@ -192,8 +196,37 @@ func (j JSONDoc) Get(key string) interface{} {
 	return j.M[key]
 }
 
-// Valid implements permissions.Validable on JSONDoc
+// Valid implements permissions.Validable on JSONDoc.
+//
+// The `referenced_by` selector is a special case: the `values` field of such
+// rule has the format "doctype/id" and it cannot directly be compared to the
+// same field of a JSONDoc since, in the latter, the format is:
+// "referenced_by": [
+//     {"type": "doctype1", "id": "id1"},
+//     {"type": "doctype2", "id": "id2"},
+// ]
 func (j JSONDoc) Valid(field, value string) bool {
+	if field == selectorReferencedBy {
+		rawReferences := j.Get(field)
+		if rawReferences == nil {
+			return false
+		}
+		references := rawReferences.([]interface{})
```

Comment:
It would be better to check the type assertion, like this:

```go
references, ok := rawReferences.([]interface{})
if !ok {
    return false
}
```